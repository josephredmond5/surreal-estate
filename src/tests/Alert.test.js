import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../components/Alert';

describe('Alert component', () => {
  it('should render null when no message is provided', () => {
    const { container } = render(<Alert />);
    expect(container.firstChild).toBeNull();
  });

  it('should match snapshot when success prop is true', () => {
    const { getByText, asFragment } = render(
      <Alert message="Success!" success />
    );
    expect(getByText(/Success/).textContent).toBe('Success!');
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot when success prop is false', () => {
    const { getByText, asFragment } = render(
      <Alert message="Error!" success={false} />
    );
    expect(getByText(/Error/).textContent).toBe('Error!');
    expect(asFragment()).toMatchSnapshot();
  });
});
