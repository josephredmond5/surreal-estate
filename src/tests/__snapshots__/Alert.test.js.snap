// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should match snapshot when success prop is false 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert component should match snapshot when success prop is true 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!
  </div>
</DocumentFragment>
`;
